### DISCUSSION TITLE
ts challenges 43

### DISCUSSION THREAD

**Discussion by maskle** [type-challenges/type-challenges#13619] [Q&A] [answered]
type MyExclude<T, U> = T extends (U | infer A) ? A : never

type test = MyExclude<'a', 'a'>

why this "test" type is "a"?   i think it should be "never"

**Comment by JonghwanWon** [comment, marked as answer]
Hi @maskle 

`T extends (U | infer A)` is always `true`.

`T` inferred to be `A` because `(U | infer A)` is union.
That is, `T` expands `T`. So it becomes true.

Hope this helps.
